pkg/objstore/s3/s3.go

```diff
@@ -246,13 +248,24 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 		client.TraceOn(logWriter)
 	}
 
+	var useListObjectsV1 bool
+	if config.ListObjectsVersion == "" || config.ListObjectsVersion == "v2" {
+		useListObjectsV1 = false
+	} else if config.ListObjectsVersion == "v1" {
+		useListObjectsV1 = true
+	} else {
+		listObjectsErrMsg := errors.Errorf("Unsupported version %q was provided. Supported values are v1, v2", config.ListObjectsVersion)
+		return nil, errors.Wrap(listObjectsErrMsg, "Initialize s3 client list objects version")
+	}
+
```

Comment:
I'm wondering if this code needs a unit test.

Reply:
Probably, but also this code can be simplified to:

```
	if config.ListObjectsVersion =! "" && c config.ListObjectsVersion =! "v1" && config.ListObjectsVersion != "v2" {
		return nil, errors.Errorf(Initialize s3 client list objects version: Unsupported version %q was provided. Supported values are v1, v2", config.ListObjectsVersion)
       }
       return Bucket{ 
           .....
           listObjectsV1:  config.ListObjectsVersion == "v1",
}
```

Reply:
Nice one, wish I wrote it that way the first time.